packages/PharBuilder/src/PharBuilder.php

```diff
@@ -0,0 +1,29 @@
+<?php declare(strict_types=1);
+
+namespace Rector\PharBuilder;
+
+use Rector\PharBuilder\Compiler\Compiler;
+
+final class PharBuilder
+{
+    /**
+     * @var Compiler
+     */
+    private $compiler;
+
+    public function __construct(Compiler $compiler)
+    {
+        $this->compiler = $compiler;
+    }
+
+    public function build(string $buildDirectory): void
+    {
+        $this->init();
+        $this->compiler->compile($buildDirectory);
+    }
+
+    private function init(): void
+    {
+        ini_set('phar.readonly', '0');
```

Comment:
I don't think this is working unless `phar.readonly` is actually already enabled

Reply:
I have no idea, just used it from some other PHAR build tool. Could you confirm? I can remove it then

Reply:
I can confirm:

```
$ psysh
>>> ini_get('phar.readonly')
=> "1"
>>> ini_set('phar.readonly', 0)
=> false
```

And [the doc](https://secure.php.net/manual/en/phar.configuration.php#ini.phar.readonly) confirms it as well:

>This setting can only be unset in php.ini due to security reasons. If phar.readonly is disabled in php.ini, the user may enable phar.readonly in a script or disable it later. If phar.readonly is enabled in php.ini, a script may harmlessly "re-enable" the INI variable, but may not disable it.

Reply:
Thanks!

Let me know if you handle that in some way